modules/renter/files.go

```diff
@@ -112,8 +112,9 @@ func (f *file) uploadProgress() float64 {
 
 // redundancy returns the redundancy of the least redundant chunk. A file
 // becomes available when this redundancy is >= 1. Assumes that every piece is
-// unique within a file contract. -1 is returned if the file has size 0.
-func (f *file) redundancy() float64 {
+// unique within a file contract. -1 is returned if the file has size 0. It
+// takes one argument, a map of offline contracts for this file.
+func (f *file) redundancy(offlineContracts map[types.FileContractID]bool) float64 {
```

Comment:
should `available()` be updated like this also, so it doesn't count pieces from offline contracts?